pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1806,7 +1802,7 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 			break
 		}
 	}
-	if needTLSInspector {
+	if needTLSInspector || opts.needHTTPInspector {
```

Comment:
Why this since `appendListenerFilters` can do that?

Reply:
`appendListenerFilters` is only used when merging conflict filter chain. This code is called when building the listener. If there is no conflict, `appendListenerFilters` will not be called.